### PR TITLE
Product listing and Filter

### PR DESCRIPTION
**Pull request by AbhishekkGautam**
### Filter and Sort
- Sort By Price
- Categories
- Brands
- Rating
- Price Range
- Cash On Delivery
- Fast Delivery
- Out of Stock
- Clear All Filters - button

### Product List Page
- product cards with info
- Out of Stock text overlay on the card.
- price after the discount ends with 99 dynamically.

### CODE REVIEW

File: src/components/ProductList/ProductList.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+              _id,
+              title,
+              brand,
+              starRating,
+              imageSrc,
+              priceInMrp,
+              discountInPercentage,
+              inStock,
+            } = product;
+            const _priceAfterDiscount = priceAfterDiscount(
```
Comment (srishti-maurya) [inline src/components/ProductList/ProductList.js:37]: I would suggest to use any one naming convention throughout

**Review by srishti-maurya — APPROVED** [review]
I liked how you have handled `loading`, `error` and the `data` in `ProductsListing` and the way you have stored all the constants of reducer

File: src/components/Navbar/Navbar.css [hunk 2]
```diff
@@ -46,7 +46,7 @@
   padding: 0;
 }
 
-.login-btn {
+.hide-on-mobile {
   display: none;
 }
```
Comment (BelideNikhil) [inline src/components/Navbar/Navbar.css:52]: try a better naming convention here , ignore if this is a utility

**Review by manojasarsa — APPROVED** [review]
The components are well structured.
Also, The code seems clean. 
Good to merge!

**Review by BelideNikhil — APPROVED** [review]
The filter and Sort button are not working on mobile view
Rest looks good!
Nice UI and clean code.

File: src/components/Navbar/Navbar.css (hunk 2 above)
Comment (AbhishekkGautam) [reply src/components/Navbar/Navbar.css:52]: Yes Nikhil, this naming is based on breakpoints.

File: src/components/ProductList/ProductList.js (hunk 1 above)
Comment (AbhishekkGautam) [reply src/components/ProductList/ProductList.js:37]: I used the underscore to differentiate between variable & function while retained the same meaning for better readability.

**Comment by AbhishekkGautam** [comment]
> The filter and Sort button are not working on mobile view Rest looks good! Nice UI and clean code.

Yes. I'll do the responsive part in a separate PR.

**Pull request merged**